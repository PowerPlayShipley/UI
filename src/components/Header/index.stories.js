import { action } from "@storybook/addon-actions";

import Header from "./index";

export default {
  title: 'Header',
  component: Header,
  argTypes: {
    as: {
      options: [null, 'h1', 'h2', 'h3', 'h4', 'h5', 'h6'],
      control: { type: 'select' }
    },
    level: {
      options: [null, 'h1', 'h2', 'h3', 'h4', 'h5', 'h6'],
      control: { type: 'select' }
    }
  }
}

const Template = (args) => {
  return <Header {...args} />
}

export const H1 = Template.bind({});
H1.args = {
  title: 'Header',
  as: 'h1'
}

export const H2 = Template.bind({});
H2.args = {
  title: 'Header',
  as: 'h2'
}

export const H3 = Template.bind({});
H3.args = {
  title: 'Header',
  as: 'h3'
}

export const H4 = Template.bind({});
H4.args = {
  title: 'Header',
  as: 'h4'
}

export const H5 = Template.bind({});
H5.args = {
  title: 'Header',
  as: 'h5'
}

export const H6 = Template.bind({});
H6.args = {
  title: 'Header',
  as: 'h6'
}

export const WithChildren = Template.bind({});
WithChildren.args = {
  children: (<button type="button" onClick={action('Clicked Children')}>Children</button>)
}
