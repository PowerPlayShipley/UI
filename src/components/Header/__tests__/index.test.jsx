/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render, screen } from '@testing-library/react'

import Header from "../index";

describe('Header', () => {
  it('should display header', () => {
    render(<Header as={'h1'} title={'Hello World'} />)
    expect(screen.getByRole('heading')).toHaveTextContent('Hello World')
  })

  it('should have `as` set to `h5`', () => {
    const { container } = render(<Header as={'h5'} title={'Hello World'} />)

    const header = container.querySelector('h5')
    expect(header).toHaveClass('h5')

    expect(container.firstChild).toMatchSnapshot()
  })
});
