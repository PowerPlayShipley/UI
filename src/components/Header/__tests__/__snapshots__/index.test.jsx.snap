// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should have \`as\` set to \`h5\` 1`] = `
<h5
  class="h5"
>
  Hello World
</h5>
`;
